FinancialTracker/Firebase.swift

```diff
@@ -68,6 +68,16 @@ class FirebaseHandler {
     }
     
     // MARK: - Methods
+    func checkAuthorisedState(completionHandler: @escaping (Bool) -> Void) {
```

Comment:
`checkAuthorisedState` returns asynchronously. While here in the `SceneDelegate`:

```swift
 FirebaseHandler.shared.checkAuthorisedState { success in
            if success {
                window.rootViewController = mainVC
            } else {
                window.rootViewController = entryVC
        }
     }
            
        self.window = window
        
        self.window?.makeKeyAndVisible()
....
```

You combine using async with sync. Looks like a bug about to show up sooner to later.